### DISCUSSION TITLE
Mail.send never ends

### DISCUSSION THREAD

**Discussion by bpingris** [adonisjs/core#1606] [Help]
Hi,

I'm using the @adonis/mail package to send mail with mailgun and I'm facing a weird issue. Here my code
```ts
await Mail.send(message => {
    message.from('mail@from.com')
      .to('mail@to.com')
      .subject('subject')
      .htmlView('emails/template', { params: 123 })
  }
)
response.ok('success')
```
The issue is that the `Mail.send` never ends and `reponse.ok` is never called. Howver the mail is well sent to the address I give.
I tried to restart of course but without success.
Do you have an idea of what the problem is ?

**Comment by bpingris** [comment]
So I tried to debug this part where I send the mail. 
This is where the application hangs forever;

```ts
// https://github.com/adonisjs/mail/blob/36fee9d60eee26a4f0e1b710d34799da8ed71703/src/Mail/Mailer.ts#L67
const response = await this.driver.send(mail.message, mail.config);
```
I have no clue of what's going on here.
I would be happy if someone knows why this line is blocking :D !

**Comment by bpingris** [comment]
When I debug in VScode, I can see what's the line that seems to block is that:
```ts
	then(onFulfilled, onRejected) {
		// eslint-disable-next-line promise/prefer-await-to-then
		return this._promise.then(onFulfilled, onRejected);
	}
```
This comes from the package [pcancelable](https://github.com/sindresorhus/p-cancelable/blob/master/index.js#L65), if that can help to find the issue

**Comment by bpingris** [comment]
Did anyone have this issue ? How can I test this behaviour more to find from where this problem comes ?
Isn't there a way to have a "debug" file and then run a `node ace debug` to run this said file ? could be faster to debug this.
Thanks

**Reply by thetutlage** [reply]
I suspect the call at the network level is taking time/getting blocked. Have you tried hitting the Mailgun API as a Curl request or using Postman?

**Comment by bpingris** [comment]
Yea I did and it works as you expect, the mails is sent without any issue on my side.
I tried again to send the mail from adonis and I still had the same behavior where the mail is sent but the code hangs forever.

**Reply by thetutlage** [reply]
Mind sharing the code in a repo for me to reproduce?

**Reply by bpingris** [reply]
Sure, here's the [repo](https://github.com/BenoitPingris/mail-issue)
There are dummy values (for addresses) in the repo but I tried with real ones and I still have the issue

**Reply by thetutlage** [reply]
Just tried and worked like a charm for me. Which version of Node.js you are using?

**Reply by bpingris** [reply]
Oh what did I do wrong :( ? I have 14.10, what's yours ?

**Reply by thetutlage** [reply]
> Oh what did I do wrong

Nothing. I just used the same code and replaced with my personal credentials. 

> I have 14.10, what's yours

I am on `v14.8.0`. Maybe trying using this version (although it should not matter)

The main issue is that the network call is getting stuck and it can be because of many reasons

**Reply by bpingris** [reply]
Just to be clear, on your side nothing was getting blocked or other weird thing, the mail has been sent and your client received a `200 cool` response ?
I'm out of idea to try to know what's the issue, a real mystery !

**Reply by thetutlage** [reply]
Yes. I received a `200 cool`. Yes, even I am confused, what can be the real issue

- Are you using the exact same code you shared with me?
- Have you tried switching networks. For example: If the issue happens in your office, then give it a try at home?

**Reply by bpingris** [reply]
Yes I used the same code, I'll try with another network. 
Did you use Mailgun to try it or did you use another service ?

**Reply by bpingris** [reply]
Ok big news, I launched a docker container with node 14.8, make my request and both the mail and the server response are ok !
I think for now I'll just dockerize my main adonis app with node 14.8
![2020-09-23-174720_360x205_scrot](https://user-images.githubusercontent.com/29386109/94036703-e3790d00-fdc4-11ea-822b-01c56f7a8df8.png)
![2020-09-23-174826_341x72_scrot](https://user-images.githubusercontent.com/29386109/94036755-f390ec80-fdc4-11ea-9204-134f85cc521c.png)

**Reply by bpingris** [reply]
Ok, false alert, I guess. I change my dockerfile to use node 14.10 (same version I have on my local computer), and it works too. So I tried again with my local node 14.10 but I still have the issue, this is so, so weird.
I will still dockerize my app but this mystery will haunt me wtf

**Reply by thetutlage** [reply]
Which operating system are you on? Maybe share this code with some of your friend and try to run it on their computer (without docker). For sure, this has to do something with the network. But I am no expert in networking, so can hardly suggest anything

**Reply by bpingris** [reply]
I'm on manjaro, I saw that adonis/mail uses nodemailer to send mails and I already used nodemailer for some projects without any issues like this one.
